packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2281,6 +2281,13 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       // "export declare" is equivalent to just "export".
       const isDeclare = this.eatContextual("declare");
 
+      if (
+        this.isContextual("declare") ||
+        !this.shouldParseExportDeclaration()
+      ) {
+        super.unexpected(this.state.start, this.state.value);
```

Comment:
Nit: we always use `this`, unless it's actually needed to use `super`.

```suggestion
        this.unexpected(this.state.start, this.state.value);
```

Reply:
🤦🏻‍♂️I missed that